libraries/core_libs/network/src/tarcap/packets_handlers/get_votes_sync_packet_handler.cpp

```diff
@@ -39,6 +39,12 @@ void GetVotesSyncPacketHandler::process(const PacketData &packet_data, const std
     std::vector<std::shared_ptr<Vote>> send_next_votes_bundle;
     for (auto &v : next_votes_bundle) {
       if (!peer->isVoteKnown(v->getHash())) {
+        if (send_next_votes_bundle.size() == kMaxVotesInPacket) {
+          sendPbftNextVotes(packet_data.from_node_id_, send_next_votes_bundle);
+          send_next_votes_bundle.clear();
+          // Allow other processing when sending large number of votes
+          thisThreadSleepForMilliSeconds(100);
```

Comment:
Sometimes there can be a lot of votes to actually send out and at the same time this could block other packets from being sent/processed like transactions and blocks. So the sleep is just to allow other processing on other threads to do its work. This is more a problem on the receiving end and not so much when we send it. We had this issue on testnet with transactions when we got stuck and we had a large transaction pool. Status message was not processed in time since we would flood the connection with transaction packet.

Reply:
I see what you want to achieve here, but sleep just blocks this thread and actually prolongs it's execution. So you are doing exact opposite. You should check how we handle trxs, and re-do it that way

Reply:
I made changes to use the periodic threadpool for this.